When the reserved keyword uses backquotes as input, deserialization causes parseSqlToExpr syntax error, which results in a checkpoint failure.

<!-- (At least include the following, feel free to add if you have more content) -->

### Steps to reproduce the behavior (Required)
CREATE TABLE `uds_nio_user_group_data` ( `group_code` varchar(256) NOT NULL COMMENT "verison code", `group_data_version` varchar(32) NOT NULL COMMENT "version", `entity_id` varchar(64) NOT NULL COMMENT "version id", `partition` date NOT NULL COMMENT "verison" ) ENGINE=OLAP DUPLICATE KEY(`group_code`, `group_data_version`, `entity_id`) PARTITION BY date_trunc('day',`partition`) DISTRIBUTED BY HASH(`entity_id`) BUCKETS 30 PROPERTIES ( "replication_num" = "3", "in_memory" = "false", "enable_persistent_index" = "false", "replicated_storage" = "true", "compression" = "LZ4" )

### Expected behavior (Required)
the fe state is normal
### Real behavior (Required)
```
2024-10-09 19:57:57.156+08:00 WARN (stateChangeExecutor|79) [GlobalStateMgr.replayJournalInner():2395] catch exception when replaying journal, id: 790, data: null,
com.starrocks.journal.JournalException: fail to read journal entity key=790, data=<DatabaseEntry offset="0" size="23373" data="50 200 0 0 91 71 123 34 100 34 58 34 116 101 115 116 34 44 34 116 34 58 123 34 99 108 97 122 122 34 58 34 79 108 97 112 84 97 98 108 101 34 44 34 99 108 117 115 116 101 114 73 100 34 58 49 49 50 49 51 54 54 56 51 51 44 34 115 116 97 116 101 34 58 34 78 79 82 77 65 76 34 44 34 105 110 100 101 120 73 100 84 111 77 101 116 97 34 58 123  ... 23273 bytes not shown "/>
        at com.starrocks.journal.bdbje.BDBJournalCursor.deserializeData(BDBJournalCursor.java:254) ~[starrocks-fe.jar:?]
        at com.starrocks.journal.bdbje.BDBJournalCursor.next(BDBJournalCursor.java:292) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.replayJournalInner(GlobalStateMgr.java:2374) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.replayJournal(GlobalStateMgr.java:2333) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.transferToLeader(GlobalStateMgr.java:1317) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.access$100(GlobalStateMgr.java:348) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr$1.transferToLeader(GlobalStateMgr.java:820) ~[starrocks-fe.jar:?]
        at com.starrocks.ha.StateChangeExecutor.runOneCycle(StateChangeExecutor.java:103) ~[starrocks-fe.jar:?]
        at com.starrocks.common.util.Daemon.run(Daemon.java:107) ~[starrocks-fe.jar:?]
Caused by: com.starrocks.sql.parser.ParsingException: Getting syntax error at line 1, column 10. Detail message: Unexpected input '(', the most similar input is {<EOF>}.
        at com.starrocks.sql.parser.ErrorHandler.syntaxError(ErrorHandler.java:40) ~[starrocks-fe.jar:?]
        at org.antlr.v4.runtime.ProxyErrorListener.syntaxError(ProxyErrorListener.java:41) ~[antlr4-runtime-4.9.2.jar:4.9.2]
        at org.antlr.v4.runtime.Parser.notifyErrorListeners(Parser.java:544) ~[antlr4-runtime-4.9.2.jar:4.9.2]
        at com.starrocks.sql.parser.StarRocksDefaultErrorStrategy.reportInputMismatch(StarRocksDefaultErrorStrategy.java:78) ~[starrocks-fe.jar:?]
        at org.antlr.v4.runtime.DefaultErrorStrategy.reportError(DefaultErrorStrategy.java:139) ~[antlr4-runtime-4.9.2.jar:4.9.2]
        at com.starrocks.sql.parser.StarRocksParser.expressionSingleton(StarRocksParser.java:39163) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.parser.SqlParser.invokeParser(SqlParser.java:251) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.parser.SqlParser.parseSqlToExpr(SqlParser.java:195) ~[starrocks-fe.jar:?]
        at com.starrocks.catalog.ExpressionRangePartitionInfo.gsonPostProcess(ExpressionRangePartitionInfo.java:95) ~[starrocks-fe.jar:?]
        at com.starrocks.persist.gson.GsonUtils$ProcessHookTypeAdapterFactory$1.read(GsonUtils.java:689) ~[starrocks-fe.jar:?]
        at com.google.gson.TypeAdapter.fromJsonTree(TypeAdapter.java:285) ~[spark-dpp-1.0.0.jar:?]
        at com.starrocks.persist.gson.RuntimeTypeAdapterFactory$1.read(RuntimeTypeAdapterFactory.java:415) ~[starrocks-fe.jar:?]
        at com.google.gson.TypeAdapter$1.read(TypeAdapter.java:199) ~[spark-dpp-1.0.0.jar:?]
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:131) ~[spark-dpp-1.0.0.jar:?]
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:222) ~[spark-dpp-1.0.0.jar:?]
        at com.starrocks.persist.gson.GsonUtils$ProcessHookTypeAdapterFactory$1.read(GsonUtils.java:687) ~[starrocks-fe.jar:?]
        at com.google.gson.TypeAdapter.fromJsonTree(TypeAdapter.java:285) ~[spark-dpp-1.0.0.jar:?]
        at com.starrocks.persist.gson.RuntimeTypeAdapterFactory$1.read(RuntimeTypeAdapterFactory.java:415) ~[starrocks-fe.jar:?]
        at com.google.gson.TypeAdapter$1.read(TypeAdapter.java:199) ~[spark-dpp-1.0.0.jar:?]
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:131) ~[spark-dpp-1.0.0.jar:?]
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:222) ~[spark-dpp-1.0.0.jar:?]
        at com.starrocks.persist.gson.GsonUtils$ProcessHookTypeAdapterFactory$1.read(GsonUtils.java:687) ~[starrocks-fe.jar:?]
        at com.google.gson.Gson.fromJson(Gson.java:963) ~[spark-dpp-1.0.0.jar:?]
        at com.google.gson.Gson.fromJson(Gson.java:928) ~[spark-dpp-1.0.0.jar:?]
        at com.google.gson.Gson.fromJson(Gson.java:877) ~[spark-dpp-1.0.0.jar:?]
        at com.google.gson.Gson.fromJson(Gson.java:848) ~[spark-dpp-1.0.0.jar:?]
        at com.starrocks.journal.JournalEntity.readFields(JournalEntity.java:274) ~[starrocks-fe.jar:?]
        at com.starrocks.journal.bdbje.BDBJournalCursor.deserializeData(BDBJournalCursor.java:248) ~[starrocks-fe.jar:?]
        ... 8 more
```
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

--- a/fe/fe-core/src/main/java/com/starrocks/analysis/SlotRef.java
+++ b/fe/fe-core/src/main/java/com/starrocks/analysis/SlotRef.java
@@ -63,8 +63,9 @@ import static com.google.common.base.Preconditions.checkArgument;
 public class SlotRef extends Expr {
     private TableName tblName;
     private String col;
-    // Used in toSql
+    // isBackQuoted/label used in toSql
     private String label;
+    private boolean isBackQuoted = false;
 
     private QualifiedName qualifiedName;
 
@@ -280,7 +281,12 @@ public class SlotRef extends Expr {
         if (tblName != null && !isFromLambda()) {
             return tblName.toSql() + "." + "`" + col + "`";
         } else if (label != null) {
-            return label;
+            if (isBackQuoted) {
+                sb.append("`").append(label).append("`");
+                return sb.toString();
+            } else {
+                return label;
+            }
         } else if (desc.getSourceExprs() != null) {
             sb.append("<slot ").append(desc.getId().asInt()).append(">");
             for (Expr expr : desc.getSourceExprs()) {
@@ -500,6 +506,14 @@ public class SlotRef extends Expr {
         SlotRef slotRef = new SlotRef();
         slotRef.readFields(in);
         return slotRef;
+    }
+
+    public void setBackQuoted(boolean isBackQuoted) {
+        this.isBackQuoted = isBackQuoted;
+    }
+
+    public boolean isBackQuoted() {
+        return isBackQuoted;
     }
 
     /**

--- a/fe/fe-core/src/main/java/com/starrocks/sql/ast/Identifier.java
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/ast/Identifier.java
@@ -22,6 +22,8 @@ public class Identifier implements ParseNode {
 
     private final NodePosition pos;
 
+    private boolean isBackQuoted = false;
+
     public Identifier(String value) {
         this(value, NodePosition.ZERO);
     }
@@ -38,6 +40,14 @@ public class Identifier implements ParseNode {
     @Override
     public NodePosition getPos() {
         return pos;
+    }
+
+    public void setBackQuoted(boolean isBackQuoted) {
+        this.isBackQuoted = isBackQuoted;
+    }
+
+    public boolean isBackQuoted() {
+        return isBackQuoted;
     }
 
     @Override

--- a/fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java
@@ -6345,7 +6345,11 @@ public class AstBuilder extends StarRocksBaseVisitor<ParseNode> {
         List<String> parts = new ArrayList<>();
         parts.add(identifier.getValue());
         QualifiedName qualifiedName = QualifiedName.of(parts, createPos(context));
-        return new SlotRef(qualifiedName);
+        SlotRef slotRef = new SlotRef(qualifiedName);
+        if (identifier.isBackQuoted()) {
+            slotRef.setBackQuoted(true);
+        }
+        return slotRef;
     }
 
     @Override
@@ -6413,7 +6417,9 @@ public class AstBuilder extends StarRocksBaseVisitor<ParseNode> {
 
     @Override
     public ParseNode visitBackQuotedIdentifier(StarRocksParser.BackQuotedIdentifierContext context) {
-        return new Identifier(context.getText().replace("`", ""), createPos(context));
+        Identifier backQuotedIdentifier = new Identifier(context.getText().replace("`", ""), createPos(context));
+        backQuotedIdentifier.setBackQuoted(true);
+        return backQuotedIdentifier;
     }
 
     @Override

--- a/fe/fe-core/src/test/java/com/starrocks/catalog/ExpressionRangePartitionInfoTest.java
+++ b/fe/fe-core/src/test/java/com/starrocks/catalog/ExpressionRangePartitionInfoTest.java
@@ -27,6 +27,7 @@ import com.starrocks.persist.gson.GsonUtils;
 import com.starrocks.qe.ConnectContext;
 import com.starrocks.server.GlobalStateMgr;
 import com.starrocks.sql.ast.CreateTableStmt;
+import com.starrocks.sql.ast.ExpressionPartitionDesc;
 import com.starrocks.sql.ast.PartitionKeyDesc;
 import com.starrocks.sql.ast.PartitionKeyDesc.PartitionRangeType;
 import com.starrocks.sql.ast.PartitionValue;
@@ -593,5 +594,116 @@ public class ExpressionRangePartitionInfoTest {
         Assert.assertTrue(slotRef.getType().isDateType());
         starRocksAssert.dropMaterializedView("test_mv1");
         starRocksAssert.dropTable("test_table");
+    }
+
+
+    @Test
+    public void testExpressionRangePartitionInfoWithReservedKeywordSerialized() throws Exception {
+        ConnectContext ctx = starRocksAssert.getCtx();
+        String createSQL = "CREATE TABLE table_reserverd_keyword_partition (\n" +
+                "databaseName varchar(200) NULL COMMENT \"\",\n" +
+                "tableName varchar(200) NULL COMMENT \"\",\n" +
+                "queryTime varchar(50) NULL COMMENT \"\",\n" +
+                "queryId varchar(50) NULL COMMENT \"\",\n" +
+                "partitionHitSum int(11) NULL COMMENT \"\",\n" +
+                "partitionSum int(11) NULL COMMENT \"\",\n" +
+                "tabletHitNum int(11) NULL COMMENT \"\",\n" +
+                "tabletSum int(11) NULL COMMENT \"\",\n" +
+                "startHitPartition varchar(20) NULL COMMENT \"\",\n" +
+                "`partition` date NULL COMMENT \"\",\n" +
+                "clusterAddress varchar(50) NULL COMMENT \"\",\n" +
+                "costTime int(11) NULL COMMENT \"\",\n" +
+                "tableQueryCount int(11) NULL COMMENT \"\"\n" +
+                ") ENGINE=OLAP\n" +
+                "DUPLICATE KEY(databaseName, tableName)\n" +
+                "COMMENT \"OLAP\"\n" +
+                "PARTITION BY date_trunc('day', `partition`)\n" +
+                "DISTRIBUTED BY HASH(databaseName) BUCKETS 1\n" +
+                "PROPERTIES (\n" +
+                "\"replication_num\" = \"1\",\n" +
+                "\"in_memory\" = \"false\",\n" +
+                "\"enable_persistent_index\" = \"false\",\n" +
+                "\"replicated_storage\" = \"true\",\n" +
+                "\"compression\" = \"LZ4\"\n" +
+                ");";
+        CreateTableStmt createTableStmt = (CreateTableStmt) UtFrameUtils.parseStmtWithNewParser(createSQL, ctx);
+        ExpressionPartitionDesc exprPartitiondesc = (ExpressionPartitionDesc) createTableStmt.getPartitionDesc();
+        FunctionCallExpr funExpr = (FunctionCallExpr) exprPartitiondesc.getExpr();
+        SlotRef slotRef = (SlotRef) funExpr.getChild(1);
+        Assert.assertTrue(slotRef.isBackQuoted());
+
+
+        StarRocksAssert.utCreateTableWithRetry(createTableStmt);
+        Database db = GlobalStateMgr.getCurrentState().getDb("test");
+        Table table = db.getTable("table_reserverd_keyword_partition");
+        ExpressionRangePartitionInfo expressionRangePartitionInfo =
+                (ExpressionRangePartitionInfo) ((OlapTable) table).getPartitionInfo();
+        String exprToSql = expressionRangePartitionInfo.getPartitionExprs().get(0).toSql();
+        Assert.assertEquals("date_trunc('day', `partition`)", exprToSql);
+        // serialize
+        String json = GsonUtils.GSON.toJson(table);
+        // deserialize
+        OlapTable readTable = GsonUtils.GSON.fromJson(json, OlapTable.class);
+        expressionRangePartitionInfo = (ExpressionRangePartitionInfo) readTable.getPartitionInfo();
+        List<Expr> readPartitionExprs = expressionRangePartitionInfo.getPartitionExprs();
+        Function fn = readPartitionExprs.get(0).getFn();
+        Assert.assertNotNull(fn);
+    }
+
+    @Test
+    public void testExpressionRangePartitionInfoWithReservedKeywordSerializedWrong() throws Exception {
+        ConnectContext ctx = starRocksAssert.getCtx();
+        String createSQL = "CREATE TABLE table_reserverd_keyword_partition1 (\n" +
+                "databaseName varchar(200) NULL COMMENT \"\",\n" +
+                "tableName varchar(200) NULL COMMENT \"\",\n" +
+                "queryTime varchar(50) NULL COMMENT \"\",\n" +
+                "queryId varchar(50) NULL COMMENT \"\",\n" +
+                "partitionHitSum int(11) NULL COMMENT \"\",\n" +
+                "partitionSum int(11) NULL COMMENT \"\",\n" +
+                "tabletHitNum int(11) NULL COMMENT \"\",\n" +
+                "tabletSum int(11) NULL COMMENT \"\",\n" +
+                "startHitPartition varchar(20) NULL COMMENT \"\",\n" +
+                "`partition` date NULL COMMENT \"\",\n" +
+                "clusterAddress varchar(50) NULL COMMENT \"\",\n" +
+                "costTime int(11) NULL COMMENT \"\",\n" +
+                "tableQueryCount int(11) NULL COMMENT \"\"\n" +
+                ") ENGINE=OLAP\n" +
+                "DUPLICATE KEY(databaseName, tableName)\n" +
+                "COMMENT \"OLAP\"\n" +
+                "PARTITION BY date_trunc('day', `partition`)\n" +
+                "DISTRIBUTED BY HASH(databaseName) BUCKETS 1\n" +
+                "PROPERTIES (\n" +
+                "\"replication_num\" = \"1\",\n" +
+                "\"in_memory\" = \"false\",\n" +
+                "\"enable_persistent_index\" = \"false\",\n" +
+                "\"replicated_storage\" = \"true\",\n" +
+                "\"compression\" = \"LZ4\"\n" +
+                ");";
+        CreateTableStmt createTableStmt = (CreateTableStmt) UtFrameUtils.parseStmtWithNewParser(createSQL, ctx);
+        ExpressionPartitionDesc exprPartitiondesc = (ExpressionPartitionDesc) createTableStmt.getPartitionDesc();
+        FunctionCallExpr funExpr = (FunctionCallExpr) exprPartitiondesc.getExpr();
+        SlotRef slotRef = (SlotRef) funExpr.getChild(1);
+        Assert.assertTrue(slotRef.isBackQuoted());
+        slotRef.setBackQuoted(false);
+
+        StarRocksAssert.utCreateTableWithRetry(createTableStmt);
+        Database db = GlobalStateMgr.getCurrentState().getDb("test");
+        Table table = db.getTable("table_reserverd_keyword_partition1");
+        ExpressionRangePartitionInfo expressionRangePartitionInfo =
+                (ExpressionRangePartitionInfo) ((OlapTable) table).getPartitionInfo();
+        String exprToSql = expressionRangePartitionInfo.getPartitionExprs().get(0).toSql();
+        Assert.assertEquals("date_trunc('day', partition)", exprToSql);
+
+        // serialize
+        String json = GsonUtils.GSON.toJson(table);
+        // deserialize
+        try {
+            OlapTable readTable = GsonUtils.GSON.fromJson(json, OlapTable.class);
+            Assert.fail();
+        } catch (Exception e) {
+            e.printStackTrace();
+            Assert.assertTrue(e.getMessage().contains("Getting syntax error at line 1, column 10. " +
+                    "Detail message: Unexpected input '(', the most similar input is {<EOF>}."));
+        }
     }
 }

